appsembler/figures
# Figures multisite support

Production code + tests for review

# Context - Support both single site and multisite deployments

The code needs to support both single site and multisite deployments. Since single site deployments do not filter "per site" in queries, we can make 'Site' FKs mandatory for Figures metrics models and assign the default site for single site deployments. The other option considered is making the Site FK optional. 

# Up for review

*  Added settings handling, tests
* Added Site to metrics models
* Pipeline is now site aware. Populate sets of metrics per site. Initially will run all sites in sequence instead of as Celery subtasks. That will be a follow-on
* Added `sites` module as an facade to retrieve site filtered data
** Add awareness for custom `edx-organizations` (namely Appsembler's fork)
* Add multisite authentication and authorization to permissions module
* Add multisite support to views


The following items should not block getting Figures installed on Tahoe staging for testing, however, we *do* want to get this test coverage as we deploy to production

* Multisite test coverage for views
* Better testing for data leaking
